File: Minimum number of Days to make m boquets/minDays.java
class Solution {
    public int minDays(int[] bloomDay, int m, int k) {
        if ((long) m * k > bloomDay.length) {
            return -1;
        }

        int low = 1;
        int high = (int) 1e9;
        while (low < high) {
            int mid = low + (high - low) / 2;
            if (canMake(bloomDay, m, k, mid)) {
                high = mid;
            } else {
                low = mid + 1;
            }
        }
        return low;
    }

    private boolean canMake(int[] bloomDay, int m, int k, int day) {
        int flowers = 0;
        int total = 0;
        for (int b : bloomDay) {
            if (b <= day) {
                flowers++;
                if (flowers == k) {
                    total++;
                    flowers = 0;
                }
            } else {
                flowers = 0;
            }
            if (total >= m) {
                return true;
            }
        }
        return false;
    }
}

File: Minimum number of Days to make m boquets/minDays.py
class Solution:
    def minDays(self, bloomDay: List[int], m: int, k: int) -> int:
        if m*k > len(bloomDay):
            return -1
        def canMake(bloomDay, m, k, day):
            total = 0
            flowers = 0 
            
            for b in bloomDay :
                if b<=day :
                    flowers += 1
                    if flowers == k :
                        total += 1 
                        flowers = 0
                else : 
                    flowers = 0
                if total >= m :
                    return True 
            return False

        low = 1
        high = max(bloomDay)
        while low < high :
            mid = (low+high)//2
            if canMake(bloomDay, m, k, mid):
                high = mid
            else :
                low = mid+1

        return low 
        